OnBoardingCalculator/app/src/main/java/com/globant/onboardingcalculator/mvp/presenter/CalculatorPresenter.java

```diff
@@ -99,8 +93,13 @@ private Double calculate() {
     }
 
     public void onEqualPressed() {
-        model.operate(String.valueOf(decimalFormat.format(calculate())));
-        view.refreshVisor(model.getFirstOperand());
+        if (model.emptyOperation()) {
+            view.showIncompletedOperationMessage();
+            view.refreshVisor(model.getFirstOperand());
```

Comment:
I think this line is not needed. if the model operation is empty shoudnt be the UI to?
If im wrong, move this line out of the if, cause is on the else branch too

Reply:
I made in this way because i want that when the equals button has been pressed (if the operator or the secondOperand be empty shows in visor the firtsOperand) and when the operation be full (fistOperand, operator and secondOperand loaded) show the result of the operation. In the furute I will change the way to show things in visor.